src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.cs

```diff
@@ -1141,6 +1141,12 @@ internal void CleanUpContainer(global::Windows.UI.Xaml.DependencyObject element)
 			else if (element is ContentControl contentControl)
 			{
 				contentControl.ClearValue(DataContextProperty);
+				if (contentControl.ContentTemplate == ItemTemplate ||
+					contentControl.ContentTemplateSelector == ItemTemplateSelector)
+				{
+					contentControl.ClearValue(ContentControl.ContentTemplateProperty);
+					contentControl.ClearValue(ContentControl.ContentTemplateSelectorProperty);
+				}
```

Comment:
@Xiaoy312 Should we be removing the `ContentTemplate` and `ItemTemplateSelector` only if each are the same as the original ones? In this case, both would get cleared if only one of them is the same.

Reply:
right... a null==null for ItemTemplateSelector would clear the ItemTemplate. that doesnt make sense.